clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosShortNamer.java

```diff
@@ -33,7 +33,7 @@
     /**
      * A pattern for parsing a auth_to_local rule.
      */
-    private static final Pattern RULE_PARSER = Pattern.compile("((DEFAULT)|(RULE:\\[(\\d*):([^\\]]*)](\\(([^)]*)\\))?(s/([^/]*)/([^/]*)/(g)?)?))");
+    private static final Pattern RULE_PARSER = Pattern.compile("((DEFAULT)|(RULE:\\[(\\d*):([^\\]]*)](\\(([^)]*)\\))?(s/([^/]*)/([^/]*)/(g)?)?))/?(L)?");
```

Comment:
Seems like "DEFAULT/L" is a valid input. Is that intentional?

Reply:
it is not intentional, but i was not using "/L" in this case.  Updated the regex to treat "DEFAULT/L" as invalid input.